### PR TITLE
[nnfw_api] Introduce expects buffer to the TrainCaseData

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces the expects buffer to store the expected data instead of using the output buffer.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft: #13062

### CODE REVIEW

**Review by ragmani — DISMISSED** [review]
LGTM

File: tests/nnfw_api/lib/GenModelTrain.h [hunk 1]
```diff
@@ -22,16 +22,34 @@
...
 struct SessionObjectTraining : public SessionObjectGeneric
 {
+  std::vector<std::vector<uint8_t>> expects;
   std::vector<std::vector<uint8_t>> losses;
 };
 
 struct TrainCaseData : public TestCaseData
 {
+  /**
+   * @brief A vector of extect buffers
```
Comment (zetwhite) [inline tests/nnfw_api/lib/GenModelTrain.h:32]:
```suggestion
   * @brief A vector of expects buffers
```

File: tests/nnfw_api/lib/GenModelTrain.h [hunk 2]
```diff
@@ -66,7 +84,7 @@ static TrainCaseData uniformTCD(const std::vector<std::vector<T>> &inputs,
   for (const auto &data : inputs)
     ret.addInput(data);
   for (const auto &data : expects)
-    ret.addOutput(data);
+    ret.addExpects(data);
```
Comment (zetwhite) [inline tests/nnfw_api/lib/GenModelTrain.h:87]: I failed to catch why this PR has been requested. 

In my understanding, this PR replaces `TestCaseData.outputs` with `TrainCaseData.expects`. 
Is there a reason this change becomes necessary?

Comment (jyoungyun) [reply tests/nnfw_api/lib/GenModelTrain.h:87]: The expected data is not the same as the output data. When this PR was introduced, the output buffers were not used. So the contributer of this change used the output buffer for the expected data. However, to calculate the accuracy in the training mode, the output buffer is required. Therefore, I introduce expects buffer for expected data.

**Review by ragmani — APPROVED** [review]

**Review by zetwhite — APPROVED** [review]
LGTM :+1:

**Pull request merged**